paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerSheetViewModel.kt

```diff
@@ -974,109 +897,54 @@ internal class CustomerSheetViewModel(
         clientSecret: String,
         paymentMethod: PaymentMethod
     ): Result<Unit> {
-        val nextStep = intentConfirmationInterceptor.intercept(
-            initializationMode = PaymentSheet.InitializationMode.SetupIntent(
-                clientSecret = clientSecret,
-            ),
-            paymentMethod = paymentMethod,
-            paymentMethodOptionsParams = null,
-            shippingValues = null,
+        intentConfirmationHandler.start(
+            arguments = IntentConfirmationHandler.Args(
+                initializationMode = PaymentSheet.InitializationMode.SetupIntent(
+                    clientSecret = clientSecret
+                ),
+                intent = stripeIntent,
+                paymentSelection = PaymentSelection.Saved(paymentMethod),
+                shippingDetails = null,
+            )
         )
 
-        unconfirmedPaymentMethod = paymentMethod
-
-        return when (nextStep) {
-            is IntentConfirmationInterceptor.NextStep.Complete -> {
+        return when (val result = intentConfirmationHandler.awaitIntentResult()) {
+            is IntentConfirmationHandler.Result.Succeeded -> {
                 safeUpdateSelectPaymentMethodState { viewState ->
-                    unconfirmedPaymentMethod?.let { method ->
-                        unconfirmedPaymentMethod = null
-
-                        viewState.copy(
-                            savedPaymentMethods = listOf(method) + viewState.savedPaymentMethods,
-                            paymentSelection = PaymentSelection.Saved(paymentMethod = method),
-                            primaryButtonVisible = true,
-                            primaryButtonLabel = resources.getString(
-                                R.string.stripe_paymentsheet_confirm
-                            ),
-                        )
-                    } ?: viewState
-                }
-                onBackPressed()
-                Result.success(Unit)
-            }
-            is IntentConfirmationInterceptor.NextStep.Confirm -> {
-                confirmStripeIntent(nextStep.confirmParams)
-                Result.success(Unit)
-            }
-            is IntentConfirmationInterceptor.NextStep.Fail -> {
-                updateViewState<CustomerSheetViewState.AddPaymentMethod> {
-                    it.copy(
-                        isProcessing = false,
-                        primaryButtonEnabled = it.formFieldValues != null,
-                        errorMessage = nextStep.message,
+                    viewState.copy(
+                        savedPaymentMethods = listOf(paymentMethod) + viewState.savedPaymentMethods,
+                        paymentSelection = PaymentSelection.Saved(paymentMethod = paymentMethod),
```

Comment:
Looks like we do `PaymentSelection.Saved(paymentMethod)` above too. Might be worth creating a variable for it.